x-pack/plugins/alerting/server/task_runner/execution_handler.ts

```diff
@@ -267,23 +278,34 @@ export class ExecutionHandler<
           continue;
         }
 
-        if (action.group === actionGroup && this.isAlertExecutable({ alertId, alert, recovered })) {
-          const state = recovered ? {} : alert.getScheduledActionOptions()?.state!;
-
-          executables.push({
-            action,
-            alert,
-            alertId,
-            actionGroup,
-            state,
-          });
+        if (action.group === actionGroup && !this.isAlertMuted(alertId)) {
+          if (
+            this.isRecoveredAlert(actionGroup) ||
+            this.isExecutableActiveAlert({ alertId, alert, action })
+          ) {
+            executables.push({
+              action,
+              alert,
+              alertId,
+              actionGroup,
+              state: alert.getScheduledActionOptions()?.state || {},
+            });
+          }
         }
       }
     }
 
     return executables;
```

Comment:
We can modify the action (by adding `lastTriggerDate`) here rather than returning the executables to update the rule later.